api/swagger.yml

```diff
@@ -610,6 +610,62 @@ components:
           type: boolean
           default: false
 
+    CommitRecordCreation:
+      type: object
+      required:
+        - version
+        - commiter
+        - message
+        - metarange_id
+        - creation_date
+        - parents
+        - metadata
+        - generation
```

Comment:
1. There are more fields here than I see [in `Commit`](https://github.com/treeverse/lakeFS/pull/7307/files#diff-ddf96e9ccf812bfbd43cf337c57640490e07f9c794ccac312f2fc4eb062c3a6bL546-L575).  How is that possible?
2. Can we just use `Commit`.. maybe doing something about the ID like ignoring it or erroring out if it does not equal the ID in the path?

Reply:
Now we have 2 problems :-)

## Some fields are missing

There is no way for the exterrnal caller of CreateCommitRecord to determine a field, if that field does not appear in the Commit record.  We will need to add them all or compute them: the caller has no way of knowing their values!

I see these fields in graveler.Commit but not in the OpenAPI spec:

- `Version`.  This field is for very old commits.  We can probably work around in the API call GetCommit: if it is CommitVersionInitial, reverse the parents list.  Now CreateCommitRecord can always create an appropriate commit.  (Oh yes, we need to get rid of the failure when the digest for the commit is different.  I am fine with reporting an error but otherwise succeeding!)
- `Generation`.  We should probably just add this as an additional field to Commit, it is not possible to compute this field.

## Too many fields are required

If a field is not required in the OpenAPI Commit response, then there is no way for the caller of CreateCommitRecord to transfer it.

Creation date occurs in OpenAPI Commit, so not an issue.  And getCommit does not promise to return commit metadata; since this is a repeated field (a map) there is no real need to pass it if empty, so don't make it required.

## Summary

I think we should:

1. Return generation from getCommit -- and make it required.
1. Make metadata _not_ required, and empty if not given.
1. Get rid of Version, always pass the parents in the correct order, and stop computing digest.  As an alternative, return Version from getCommit -- and make it required.